include/revng/ADT/ReversePostOrderTraversal.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/PostOrderIterator.h"
+
+namespace llvm {
+
+template <class GraphT, class GT = llvm::GraphTraits<GraphT>,
+          class SetType = std::set<typename llvm::GraphTraits<GraphT>::NodeRef>>
+class ReversePostOrderTraversalExt {
```

Comment:
This comes from `revng-c`, it returns the nodes in `RPOT` order of a visit that start from a set of nodes instead of a single node which is useful when you have `DAGs` or you only need a subgraph.
It is similar to `ReversePostOrderTraversal` from llvm which works well for a single entry node only